source/mainnet/smart-contracts/tutorials/piggy-bank/testing.rst

```diff
@@ -88,137 +110,290 @@ Also, make sure to have a text editor setup to write Rust.
+       let mut chain = Chain::new();
+   }
 
-.. note::
+Next step is to create two |Account|_ entities and add them to the ``chain``.
+The simplest way to create an account is with |Account_new|_, which takes an |AccountAddress|_ and a total balance of the account.
+Once constructed, use the |Chain_create_account|_ method to add it to the chain.
+This step is important, as simply constructing an |Account|_ does not make the chain aware of it.
+
+Use the addresses and initial balance defined as constants:
+
+.. code-block:: rust
 
-   As you will see later with the |TestReceiveContext|_, these placeholders have
-   setter functions, allowing us to partially specify the context.
+   // .. Imports and constants omitted for brevity
+
+   #[test]
+   fn test_init() {
+       let mut chain = Chain::new();
+       let account_owner = Account::new(ACC_ADDR_OWNER, ACC_INITIAL_BALANCE);
+       let account_other = Account::new(ACC_ADDR_OTHER, ACC_INITIAL_BALANCE);
+       chain.create_account(account_owner);
+       chain.create_account(account_other);
+   }
 
-Now you can call ``piggy_init`` and get a result containing the initial state.
+The balance of the accounts matter when using the testing library, as the cost of transactions, for example deploying a smart contract, will be deducted from the balance of the account sending the transaction.
+
+
+With the accounts created, you are ready to load and deploy the smart contract module.
+|concordium-smart-contract-testing|_ uses the compiled smart contract modules that you also deploy on the block chain.
+Use ``cargo concordium``, which you installed in preparation for this tutorial, to compile the piggy bank to web assembly (wasm).
+
+Open a terminal and use ``cd``, short for *change directory*, to go into the root of your piggy bank project.
+Then compile your contract with:
+
+.. code-block:: console
+
+   $cargo concordium build --out piggy_bank_part2.wasm.v1
+
+This produces the wasm module ``piggy_bank_part2.wasm.v1`` in the root of your project.
+That is, unless you have any typos or bugs in your code.
+If that is the case, try to fix them using the helpful error messages from the compiler or go back to the end of part 1 and copy the full contract code again.
+
+Going back to your test case, use the function |module_load_v1|_ to load the module.
 
 .. code-block:: rust
 
-   let state_result = piggy_init(&ctx, &mut state_builder);
+   // .. Imports and constants omitted for brevity
 
-First, you want the test to fail if the contract did not result in an
-initial state:
+   #[test]
+   fn test_init() {
+       // .. lines omitted for brevity.
+       let module = load_module_v1("piggy_bank_part2.wasm.v1").expect("Module is valid and exists");
+   }
+
+|module_load_v1|_ loads a module on disk, which might be missing or invalid, and it thus returns a ``Result`` type.
+You can use ``unwrap``, or ``expect`` (which you provide a helpful error message), to extract the actual module from the ``Result``.
```

Comment:
I don't understand the part in parentheses.

Reply:
I've rephrased it now.